firebase-firestore/ktx/src/test/kotlin/com/google/firebase/firestore/ktx/NestedMapEncoderTests.kt

```diff
@@ -0,0 +1,92 @@
+package com.google.firebase.firestore.ktx
+
```

Comment:
Check `MapperTest` from the java sdk for ideas of the scenarios to test. YOu may not need ton include so much detail as the java counterpart, but there are interesting cases anyway.